pkg/cloudprovider/providers/azure/azure_test.go

```diff
@@ -405,7 +440,7 @@ func getTestInternalFipConfigurationProperties(expectedSubnetName *string) netwo
 	}
 }
 
-func getTestService(identifier string, proto v1.Protocol, requestedPorts ...int32) v1.Service {
+func getTestService(identifier string, proto v1.Protocol, loadBalancerIP string, requestedPorts ...int32) v1.Service {
```

Comment:
I'm not comfortable with the tests presuming that a `loadBalancerIP` will always be provided on the `service.Spec`.  We want to be able to test correct behaviour when the IP address is dynamically assigned.  I guess for most test scenarios today, we need to provide the IP address if we care about it, and we can always pass `""` here if we explicitly want not to set a `service.Spec.LoadBalancerIP`.